test/test.c

```diff
@@ -27951,18 +27951,35 @@ test_JetStreamSubscribeIdleHearbeat(void)
     s = js_Publish(NULL, js, "foo", "msg2", 4, NULL, &jerr);
     testCond((s == NATS_OK) && (jerr == 0));
 
+    // Cheat by pretending that the server sends message seq 3, while client
+    // received only seq 1. Disable auto-ack for this message, or we mess up the
+    // server state.
+#define PUBLISH_FAKE_JS_MSG_WITH_SEQ(_reply, _msg)                         \
+    {                                                                      \
+        natsSub_Lock(sub);                                                 \
+        inbox = sub->subject;                                              \
+        sub->jsi->ackNone = true;                                          \
+        natsSub_Unlock(sub);                                               \
+                                                                           \
+        natsConn_setFilterWithClosure(nc, _setMsgReply, (void *)(_reply)); \
+        s = natsConnection_PublishString(nc, inbox, (_msg));               \
+    }
+
+#define PUBLISH_FAKE_RESET()      \
+    {                             \
+        natsSub_Lock(sub);        \
+        sub->jsi->ackNone = true; \
```

Comment:
did you mean to say `false` here?

Reply:
yes, TY. Wonder that it still passses with no auto-acks.